test/Crypto.purs

```diff
@@ -0,0 +1,58 @@
+module Test.Crypto (suite) where
+
+import Prelude
+
+import Contract.Prim.ByteArray (byteArrayToHex)
+import Control.Monad.Error.Class (class MonadThrow, throwError)
+import Data.Maybe (Maybe(..))
+import Data.Traversable (traverse_)
+import Data.Tuple (Tuple(Tuple))
+import Effect.Aff.Class (liftAff)
+import Effect.Exception (error)
+import Mote (group, test)
+import QueryM (QueryM, runQueryM, traceQueryConfig)
+import QueryM.Crypto (plutusHash, HashMethod(..))
+import Test.Spec.Assertions (shouldEqual)
+import TestM (TestPlanM)
+import Types.ByteArray (ByteArray, hexToByteArray)
+
+suite :: TestPlanM Unit
+suite = do
+  group "Contract.Crypto"
+    $ test "plutusHash"
+    $ do
+        qcf <- liftAff $ traceQueryConfig
+        liftAff $ runQueryM qcf $ do
+          mkReqs sourceHashPairs
+  where
+  mkReqs
+    :: Array (Tuple (Maybe ByteArray) (Array (Tuple HashMethod String)))
+    -> QueryM Unit
+  mkReqs = traverse_
+    ( \(Tuple origBS hashtups) ->
+        traverse_
+          ( \(Tuple meth goal) -> do
+              qbs <- throwMaybe (error "bytestring is not hex format") origBS
+              res <- throwMaybe (error "hashing method was not successfull") =<<
+                plutusHash meth qbs
+              shouldEqual goal $ byteArrayToHex res
+          )
+          hashtups
+    )
+
+throwMaybe :: forall a m e. MonadThrow e m => e -> Maybe a -> m a
```

Comment:
You can also use `Test.Utils.errMaybe` and skip using `error` explicitly above

On a general note: we always use kind signatures on type parameters, even when this can be clearly inferred (so we would even explicitly add `(m :: Type -> Type)` even though it's obvious from the `MonadThrow` constraint)

Reply:
Thank you, noted.